### ISSUE TITLE
Invalid prop: type check failed for prop "width". Expected Number, got String.

### ISSUE THREAD

**Issue by MadReal** [euvl/vue-js-toggle-button#40]
`Invalid prop: type check failed for prop "width". Expected Number, got String.`

Hi, I found this "bug" that I think happens when you are using something like VuexPersistedState to save caches. I've had the same issue when passing props to components.

The workaround I found was allowing 2 types in the props:
```
 props: {
    width: [String, Number]
}
```
but I'm not sure, maybe you don't like this solution.

Alternatively you call call the props in the component with a bind and it prevents the error in the console.
`						toggle-button(@change="collapsed = !collapsed" :value="collapsed" :labels="{checked: 'Collapsed', unchecked: 'Extended'}" color="#7322F7" :sync="true" :height="30" :width="92")
`

Maybe you have a better solution to this, just thought I'd let you know. 
Have a good day.

**Comment by euvl** [comment]
its not a but its a correct type check. 
I am afraid you will have to fix your environment rather than me adjusting a plugin to your usecase

**Comment by d14na** [comment]
> its not a bug - its a correct type check.
> I am afraid you will have to fix your environment rather than me adjusting a plugin to your usecase

I fail to see how this does not qualify as a bug. as @MadReal states, in certain VueJs environments, this fails the type check, even though you enter an integer value. I'm also confused as to why you check for `Number` when all values (including booleans) are entered with double-quotes (as shown in the README examples). After cloning and using @MadReal suggestion of `width: [String, Number]`, it works as expected in our environment as well.

Nevertheless, this is a fantastic library and we're very appreciative for it.  
Cheers!

**Comment by euvl** [comment]
Thanks, @d14na.

It is not a bug, because of string type != number. The fact that it is working with strings, does not guarantee deterministic behavior of the plugin. For example, in javascript you can serialize anything into a string, it does not mean though that you would expect an object to behave the same as string, right? But somehow you would expect string and number to behave the same way.

The point i am trying to make is that even though the values like `22` and `"22"` look the same - they are not the same. Types are types for a reason, you cannot neglect this concept for the sake of developers' convenience. It always leads to bugs, low code quality, and wrong reasoning about behavior.

**Comment by d14na** [comment]
> The point i am trying to make is that even though the values like `22` and `"22"` look the same - they are not the same.

agreed! the first thing I did when `width: "75"` threw the error was to try `width: 75`; but that also failed, claiming `type String`, when it clearly isn't.

I did take a quick look at the code, but it didn't seem apparent to me what the cause was. add to that, I'm not all that comfortable yet with webpack-like bundling; so who KNOWS what else is going on under the hood.

---

but the fact remains that it doesn't work in some environments, at least ours and the op. i'd imagine there were probably others too; but perhaps they never discovered this workaround and just found another (less desirable) library to satisfy their needs.

if i find the time, troubleshooting this could be a great exercise in helping raise my webpack-fu, which is fairly week atm
